source/ui/IDEAS/IDEASXRFDetailedDetectorViewWithSave.cpp

```diff
@@ -253,3 +255,15 @@ void IDEASXRFDetailedDetectorViewWithSave::onDeadTimeCheckButtonClicked()
 
 	deadTimeCheckActions->start();
 }
+
+void IDEASXRFDetailedDetectorViewWithSave::cleanupScanAction()
+{
+	acquireButton_->setEnabled(true);
+	saveScanButton_->setText("Scan Saved");
+	disconnect(scanAction_, SIGNAL(cancelled()), this, SLOT(cleanupScanAction()));
+	disconnect(scanAction_, SIGNAL(succeeded()), this, SLOT(cleanupScanAction()));
+	disconnect(scanAction_, SIGNAL(failed()), this, SLOT(cleanupScanAction()));
+
+	scanAction_->scheduleForDeletion();
+	scanAction_ = 0;
```

Comment:
I wasn't sure if there was a mechanism in the ScanAction system for the deletion of scans once they are complete, or whether it was the responsibility of the creator. I added a cleanup mechanism here which should return the detector view to the state it was in before it started, but if I'm doubling up duty here for something that the ScanAction system already takes care of let me know and I'll get rid of these.